### PR TITLE
new scriptaction stuff

### PR DESCRIPTION
**Pull request by Excuze**
Simple actions, I made it during school on a school laptop that hits 20 fps so don't judge me ;-;

Create Dict now has two optional list arguments to initialize as keys/values like in DiamondFire

Added If Matches Regex to text actions

Added Append List Values - copies and appends the contents of one list to another. Also from DiamondFire

### CODE REVIEW

File: src/main/java/io/github/codeutilities/script/action/ScriptActionType.java [hunk 1]
```diff
@@ -387,6 +387,22 @@ public enum ScriptActionType {
             ctx.context().setVariable(ctx.variable("List").name(), new ScriptListValue(list));
         })),
 
+    APPEND_LIST_VALUES(builder -> builder.name("Append List Values")
+            .description("Appends one list's contents to another.")
+            .icon(Items.BLAST_FURNACE)
+            .category(ScriptActionCategory.LISTS)
+            .arg("Base List", ScriptActionArgumentType.VARIABLE)
+            .arg("Other List", ScriptActionArgumentType.VARIABLE)
```
Comment (BlazeMCworld) [inline src/main/java/io/github/codeutilities/script/action/ScriptActionType.java:395]: should be ScriptActionArgumentType.LIST for "Other List"

File: src/main/java/io/github/codeutilities/script/action/ScriptActionType.java [hunk 2]
```diff
@@ -568,8 +599,25 @@ public enum ScriptActionType {
...
-            ctx.context().setVariable(ctx.variable("Result").name(), new ScriptDictionaryValue(new HashMap<>()));
+
+            HashMap<String, ScriptValue> dict = new HashMap<String, ScriptValue>();
+
+                if (ctx.argMap().containsKey("Keys") && ctx.argMap().containsKey("Values")) {
+                    List<ScriptValue> keys = ctx.value("Keys").asList();
+                    List<ScriptValue> values = ctx.value("Values").asList();
+
+                    // make sure we don't iterate past the end of a list
+                    int lowerLength = (keys.size() < values.size()) ? keys.size() : values.size();
```
Comment (BlazeMCworld) [inline src/main/java/io/github/codeutilities/script/action/ScriptActionType.java:613]: Math.min() is a thing btw

**Review by BlazeMCworld — CHANGES_REQUESTED** [review]

**Comment by Excuze** [comment]
will do these once exam is over probably

File: src/main/java/io/github/codeutilities/script/action/ScriptActionType.java (hunk 2 above)
Comment (Excuze) [reply src/main/java/io/github/codeutilities/script/action/ScriptActionType.java]: changed

**Review by TechStreetDev — APPROVED** [review]
LGTM

**Review by BlazeMCworld — APPROVED** [review]

**Pull request merged**